File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.py
def solution(n, m, section):
    answer = 0
    sec = len(section) - 1
    left = 0
    right = 1
    
    while 1:
        if left > sec:
            break
        elif left >= sec:
            answer += 1
            break
        if right > sec:
            answer += 1
            break
        
        if section[right] - section[left] > m - 1:
            answer += 1
            left = right
            right += 1
        elif section[right] - section[left] == m - 1:
            answer += 1
            left = right + 1
            right = left + 1
        else:
            right += 1
        
    return answer

File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.java
class Solution {
    public int solution(int n, int m, int[] section) {
        int answer = 0;
        int secLength = section.length;
        int start = 0;
        int end = 1;
        int paint;
        if (m == 1) {
            return secLength;
        }
        while (secLength > start) {
            if (secLength <= end) {
                answer++;
                break;
            }
            paint = section[end] - section[start] + 1;
            if (paint > m) {
                answer++;
                start = end;
                end = start + 1;
            } else if (paint == m) {
                answer++;
                start = end + 1;
                end = start + 1;
            } else {
                end++;
            }
        }
        return answer;
    }
}